pkg/controller/clusterdeployment/clusterdeployment_created.go

```diff
@@ -68,21 +72,45 @@ func (r *ReconcileClusterDeployment) handleCreate(request reconcile.Request, ins
 		return reconcile.Result{}, err
 	}
 
-	r.reqLogger.Info("Creating syncset")
-	newSS := kube.GenerateSyncSet(request.Namespace, request.Name, pdIntegrationKey)
-	if err = controllerutil.SetControllerReference(instance, newSS, r.scheme); err != nil {
-		r.reqLogger.Error(err, "Error setting controller reference on syncset")
-		return reconcile.Result{}, err
-	}
-	if err := r.client.Create(context.TODO(), newSS); err != nil {
-		if errors.IsAlreadyExists(err) {
-			// SyncSet already exists, we should just update it
-			if updateErr := r.client.Update(context.TODO(), newSS); updateErr != nil {
+	//add secret part
+	sc := &corev1.Secret{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: config.PagerDutySecretName, Namespace: instance.Namespace}, sc)
+	secret := kube.GeneratePdSecret(instance.Namespace, config.PagerDutySecretName, pdIntegrationKey)
+	if err != nil {
```

Comment:
The value of `err` here is for the Get on line 77, not for the create.  If the secret does exist, may want to delete it first.  It could have a bad integration key if it wasn't successfully deleted before.

Also this logic is very similar to the syncset_deleted logic.  Recommend pull this into a utility and reuse.

Reply:
> It could have a bad integration key if it wasn't successfully deleted before.
Could you tell my why this will happen? the secret has a ownerReference point to the clusterDeployment object, when the cd is deleted, the secret should be deleted finally.

If we have someone manually delete the syncset, not the secret, this will make sure the syncset is recreated, not the secret.

Reply:
If the PD service is deleted the key will be out of sync when the PD service is recreated.

Reply:
@jewzaam I am try to delete the secret when the pdIntegrationKey is not the same with the od secret, for a reason:
I am watching the "secret" object for the clusterDeployment reconcile, because I'd like the secret being recreated when it is deleted by mistake, https://github.com/openshift/pagerduty-operator/pull/72/files#diff-d93dafa2cfa94d56f07b0a471d5a5494R84
If we keeps delete the secret anytime when we start the sync the object, the delete of object will trigger a reconcile immedially, and running into a dead loop like this:

```
{"level":"info","ts":1572423021.678009,"logger":"pagerduty_cd","msg":"creating pd secret","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423021.684059,"logger":"pagerduty_cd","msg":"the pd secret exist, delete it first","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423021.924095,"logger":"pagerduty_cd","msg":"Creating syncset","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423021.924166,"logger":"pagerduty_cd","msg":"Creating configmap","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423021.933366,"logger":"pagerduty_cd","msg":"Reconciling ClusterDeployment","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423021.933404,"logger":"pagerduty_cd","msg":"continue","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.311224,"logger":"pagerduty_cd","msg":"creating pd secret","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.3175979,"logger":"pagerduty_cd","msg":"the pd secret exist, delete it first","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.526062,"logger":"pagerduty_cd","msg":"Creating syncset","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.5261,"logger":"pagerduty_cd","msg":"Creating configmap","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.539117,"logger":"pagerduty_cd","msg":"Reconciling ClusterDeployment","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.539156,"logger":"pagerduty_cd","msg":"continue","Request.Namespace":"haowang-test","Request.Name":"foo"}
{"level":"info","ts":1572423022.943372,"logger":"pagerduty_cd","msg":"creating pd secret","Request.Namespace":"haowang-test","Request.Name":"foo"}
```
If you didn't like the logic here, I can change to always delete the secret first if it exists, and remove the watch for the "secret", but it will never be created if someone delete the secret.